### PR TITLE
Respect conditional runs for children pjs

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/kubernetes/test-infra/issues/4504

/assign @spxtr

### CODE REVIEW

File: prow/jenkins/controller_test.go [hunk 1]
```diff
@@ -26,26 +26,60 @@ import (
...
+					Name: "push-image",
+				},
+			},
+		},
+		{
+			Name: "test-bazel-test",
+		},
+	}
+	if err := config.SetRegexes(presubmits); err != nil {
+		panic(err)
```
Comment (spxtr) [inline prow/jenkins/controller_test.go]: `t.Fatal`, here and elsewhere.

File: prow/jenkins/controller.go [hunk 2]
```diff
@@ -306,3 +311,36 @@ func (c *Controller) syncNonPendingJob(pj kube.ProwJob, reports chan<- kube.Prow
 	}
 	return nil
 }
+
+// TODO: Collapse with plank, impossible to reuse as is due to the interfaces.
+func RunAfterSuccessCanRun(parent, child *kube.ProwJob, c configAgent, ghc githubClient) bool {
```
Comment (spxtr) [inline prow/jenkins/controller.go]: Please document what this does.

File: prow/jenkins/controller_test.go (hunk 1 above)
Comment (kargakis) [reply prow/jenkins/controller_test.go]: done

File: prow/jenkins/controller.go (hunk 2 above)
Comment (kargakis) [reply prow/jenkins/controller.go]: done

File: prow/config/config.go [hunk 3]
```diff
@@ -227,7 +227,7 @@ func parseConfig(c *Config) error {
 	return nil
 }
 
-func setRegexes(js []Presubmit) error {
+func SetRegexes(js []Presubmit) error {
```
Comment (spxtr) [inline prow/config/config.go]: Please add a quick doc string for this.

**Comment by spxtr** [comment]
/lgtm
One nit.

File: prow/config/config.go (hunk 3 above)
Comment (kargakis) [reply prow/config/config.go]: Done.

**Pull request merged**